kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PersonKontaktInfoMapper.java

```diff
@@ -1,35 +1,163 @@
 package no.difi.meldingsutveksling.ptp;
 
+import com.google.common.base.MoreObjects;
 import no.difi.ptp.sikkerdigitalpost.Person;
-import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
-
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.StringWriter;
-import java.security.cert.CertificateException;
-import java.security.cert.CertificateFactory;
-import java.security.cert.X509Certificate;
+import no.difi.ptp.sikkerdigitalpost.Reservasjon;
+import no.difi.ptp.sikkerdigitalpost.Status;
+import no.difi.ptp.sikkerdigitalpost.Varslingsstatus;
+
 import java.util.Optional;
 
-public class PersonKontaktInfoMapper {
+class PersonKontaktInfoMapper {
 
-    public PersonKontaktInfoMapper() {
+    private PersonKontaktInfoMapper() {
     }
 
     static Optional<KontaktInfo> map(Person person) {
-        return Optional.of(new KontaktInfo(pemCertificateFrom(person.getX509Sertifikat()), person.getSikkerDigitalPostAdresse().getPostkasseleverandoerAdresse(), person.getSikkerDigitalPostAdresse().getPostkasseadresse(), person.getKontaktinformasjon().getEpostadresse().getValue(), person.getKontaktinformasjon().getMobiltelefonnummer().getValue(), person.getVarslingsstatus().value()));
+        PersonWrapper pw = new PersonWrapper(person);
+
+        String pemCertificate = "";
+
+
+        if(pw.hasCertificate()) {
+            pemCertificate = Util.pemCertificateFrom(pw.getX509Sertifikat());
+        }
+        MailboxProvider providerDetails = MailboxProvider.EMPTY;
+        if(pw.hasMailbox()) {
+            providerDetails = new MailboxProvider(pemCertificate,
+                    person.getSikkerDigitalPostAdresse().getPostkasseleverandoerAdresse(),
+                    person.getSikkerDigitalPostAdresse().getPostkasseadresse()
+            );
+        }
+
+        PersonDetails personDetails = PersonDetails.EMPTY;
+        if (hasKontaktInformasjon(person)) {
+            personDetails = new PersonDetails(person.getKontaktinformasjon().getEpostadresse().getValue(),
+                    person.getKontaktinformasjon().getMobiltelefonnummer().getValue(),
+                    person.getVarslingsstatus() == Varslingsstatus.KAN_VARSLES,
+                    person.getReservasjon() == Reservasjon.JA,
+                    person.getStatus() == Status.AKTIV);
+        }
+
+
+
+        return Optional.of(new KontaktInfo(providerDetails, personDetails));
+    }
+
+    private static boolean hasKontaktInformasjon(Person person) {
+        return person.getKontaktinformasjon() != null;
+    }
+
+    private static boolean hasPostkasseleverandoerAdresse(Person person) {
+        return person.getSikkerDigitalPostAdresse() != null;
+    }
+
+    private static boolean hasCertificate(Person person) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "hasCertificate" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)